lib/crypto/src/crypto.erl

```diff
@@ -297,6 +301,52 @@ strong_rand_bytes(Bytes) ->
 strong_rand_bytes_nif(_Bytes) -> ?nif_stub.
 
 
+rand_seed() ->
+    rand:seed(rand_seed_s()).
+
+rand_seed_s() ->
+    {#{ type => crypto,
+        max => infinity,
+        next => fun rand_plugin_next/1,
+        uniform => fun rand_plugin_uniform/1,
+        uniform_n => fun rand_plugin_uniform/2,
+        jump => fun rand_plugin_jump/1},
+     no_seed}.
+
+rand_plugin_next(Seed) ->
+    {bytes_to_integer(strong_rand_range(1 bsl 64)), Seed}.
+
+rand_plugin_uniform(State) ->
+    {strong_rand_float(), State}.
+
+rand_plugin_uniform(Max, State) ->
+    {bytes_to_integer(strong_rand_range(Max)) + 1, State}.
+
+rand_plugin_jump(State) ->
+    State.
+
+strong_rand_range(Range) when is_integer(Range), Range > 0 ->
+    BinRange = int_to_bin(Range),
+    strong_rand_range(BinRange);
+strong_rand_range(BinRange) when is_binary(BinRange) ->
+    case strong_rand_range_nif(BinRange) of
+        false ->
+            erlang:error(low_entropy);
+        <<BinResult/binary>> ->
+            BinResult
+    end.
+strong_rand_range_nif(_BinRange) -> ?nif_stub.
+
+strong_rand_float() ->
```

Comment:
Indeed,  I ended up rewriting it using a similar approach based on both your and @okeuday 's suggestions.